cpp/arcticdb/version/version_map.hpp

```diff
@@ -879,9 +875,13 @@ class VersionMapImpl {
                                                             std::optional<AtomKey> first_key_to_tombstone = std::nullopt,
                                                             std::shared_ptr<VersionMapEntry> entry = nullptr) {
```

Comment:
I am starting to wonder if the `first_key_to_tombstone` argument is safe before this change even @willdealtry 

Calls using it basically assumes the version map has not changed without checking the timeout.